tools/json2bin/json2bin.py

```diff
@@ -9,11 +9,14 @@
 
 ARGPARSER = ArgumentParser(
     prog='json2bin.py',
-    description='Tool for converting JSON documents to binary data via a constructed parsing schema'
+    description='Tool for converting a collection of JSON documents into\na NARC via a constructed parsing schema'
 )
-ARGPARSER.add_argument('-b', '--batch-mode', action='store_true')
-ARGPARSER.add_argument('-i', '--input', required=True)
-ARGPARSER.add_argument('-o', '--output')
+ARGPARSER.add_argument('--knarc', required=True,
```

Comment:
Don't remember if I already expresses my opinion on the matter, but I really dislike calling knarc from other tools. If we don't want to have two separate steps (i.e. json -> bins, and bins -> NARC), I'd rather do what the final rom target does (i.e. bakes several steps into one custom target by calling the shell executable and passing it the commands).

Reply:
csv2bin basically already does this chain implicitly, just with its own internal narc implementation. What's the difference for you?

Reply:
The difference is knarc is a separate tool, so if we were to make any breaking changes to how it works, or even replace it with something else, we'd have to also update all the other tools that are calling it.

Whereas if you have two separate steps, changing knarc will just require you to modify the build files.

Reply:
Can I make a proposition, then, that we accept this as a stopgap and update json2bin to have an operable narc mode that compiles the narc, like csv2bin? Would that be amenable? I can file an issue for it.

Reply:
I also want to add that csv2bin having its own NARC implementation is an issue of its own: we should have only one common NARC implementation across all tools.

Reply:
> Can I make a proposition, then, that we accept this as a stopgap and update json2bin to have an operable narc mode that compiles the narc, like csv2bin? Would that be amenable? I can file an issue for it.

Yes, I'll definitely take the current setup as a stopgap. I also really want to see this PR merged. 
Definitely file an issue to discuss about what we should do with NARC implementations.

Reply:
Done: https://github.com/pret/pokeplatinum/issues/53